util/src/main/java/com/google/cloud/healthcare/imaging/dicomadapter/monitoring/MonitoringService.java

```diff
@@ -0,0 +1,175 @@
+package com.google.cloud.healthcare.imaging.dicomadapter.monitoring;
+
+import com.google.api.Metric;
+import com.google.api.MetricDescriptor;
+import com.google.api.MonitoredResource;
+import com.google.api.client.http.HttpRequestFactory;
+import com.google.cloud.healthcare.imaging.dicomadapter.GcpMetadataUtil;
+import com.google.cloud.monitoring.v3.MetricServiceClient;
+import com.google.monitoring.v3.CreateTimeSeriesRequest;
+import com.google.monitoring.v3.Point;
+import com.google.monitoring.v3.ProjectName;
+import com.google.monitoring.v3.TimeInterval;
+import com.google.monitoring.v3.TimeSeries;
+import com.google.monitoring.v3.TypedValue;
+import com.google.protobuf.Timestamp;
+import com.google.protobuf.util.Timestamps;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.Executors;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.TimeUnit;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class MonitoringService {
+
+  private static final int DELAY = 60;
+
+  private static final String META_PROJECT_ID = "project/project-id";
+  private static final String META_LOCATION = "instance/zone";
+  private static final String META_CLUSTER_NAME = "instance/attributes/cluster-name";
+
+  private static Logger log = LoggerFactory.getLogger(MonitoringService.class);
+  private static MonitoringService INSTANCE;
+
+  private final MetricServiceClient client;
+  private final ScheduledExecutorService service;
+  private final HashMap<IMonitoringEvent, Long> aggregateEvents;
+  private final IMonitoringEvent[] monitoredEvents;
+  private final MonitoredResource monitoredResource;
+
+  private String projectId;
+
+  private MonitoringService(String projectId, IMonitoringEvent[] monitoredEvents,
+      HttpRequestFactory requestFactory) {
+    try {
+      client = MetricServiceClient.create();
+    } catch (IOException e) {
+      throw new RuntimeException(e);
+    }
+
+    aggregateEvents = new HashMap<>();
+    service = Executors.newSingleThreadScheduledExecutor();
+    service.scheduleWithFixedDelay(MonitoringService.this::flush,
+        DELAY, DELAY, TimeUnit.SECONDS);
+
+    if (projectId != null && projectId.length() > 0) {
+      this.projectId = projectId;
+    } else {
+      this.projectId = GcpMetadataUtil.get(requestFactory, META_PROJECT_ID);
```

Comment:
It can be run anywhere when using global monitored resource. 
I think projectId in particular should available for both (and it can be set explicitly via command line parameter), though I only ran adapter on GKE.

Reply:
Well, I run adapter on container optimized OS - it starts with monitoring set to global resource as expected (since I didn't and probably can't configure necessary environment variables for k8s_container resource on GCE). 
Didn't fully configure service accounts/ports to actually access it from outside, but I don't see a potential reason for it not being usable on GCE.

Reply:
Graph would be created, but with global resource (there is one global resource per project).

Setting up monitored resource for it as 'k8s_container' would be impossible (because it isn't gke and thus I won't have some necessary fields).
Setting up monitored resource as 'gce_container' is possible. Since I'm not sure what I should use as indication of raw gce vs gke on gce, it could be done as fallback.

Reply:
Ok, if the graph is created with global resource that's good.